File: 2556.java
import java.util.*;
public class Main
{
 public static void main(String[] args)
 {
  Scanner in=new Scanner(System.in);
  while(in.hasNext())
  {
   String s=in.next();
   System.out.println("300 420 moveto");
   System.out.println("310 420 lineto");
   int b=0;
   int x=310;
   int y=420;
   for(int i=0;i< s.length();i++)
    {
	char c=s.charAt(i);
	if(b==0)
	{
	   if(c=='A'){
	     y-=10;
	     b=1;
	   }
	   else {
	    y+=10;
	    b=3;
	   }
         }
	else if(b==1)
	 {
	   if(c=='A'){
	     x-=10;
	     b=2;
	   }
	   else {
	    x+=10;
	    b=0;
	   }
	}
	else if(b==2)
	 {
	   if(c=='A'){
	     y+=10;
	     b=3;
	   }
	   else {
	    y-=10;
	    b=1;
	   }
	}
	else if(b==3)
	 {
	   if(c=='A'){
	     x+=10;
	     b=0;
	   }
	   else {
	     x-=10;
	     b=2;
	   }
	}
	System.out.println(x+" "+y+" lineto");
     }
     System.out.println("stroke");
     System.out.println("showpage");
    }
 }
}

File: 2556.cc
#include<cstdio>
#include<cstring>
using namespace std;
int dr[][2]={10,0,0,-10,-10,0,0,10};
int main()
{
    char s[250];
    while(gets(s))
    {
        int x=310,y=420,d=0;
        printf("300 420 moveto\n310 420 lineto\n");
        for(int i=0;s[i]!='\0';i++)
        {
            if(s[i]=='A')
                d=(d+1)%4;
            else
                d=(d+3)%4;
            x+=dr[d][0];
            y+=dr[d][1];
            printf("%d %d lineto\n",x,y);
        }
        printf("stroke\nshowpage\n");
    }
    return 0;
}




